File: 이전/2024/7월/0705/set_list.py
import time

n = 1000000

# 리스트 곱셈 사용
start = time.time()
visited1 = [0] * n
end = time.time()
print(f'리스트 곱셈 사용: {end - start}초')

# for문 사용
start = time.time()
visited2 = []
for i in range(n):
    visited2.append(0)
end = time.time()
print(f'for문 사용: {end - start}초')

File: 이전/2024/7월/0705/setList.js
let n = 1000000;
let start, end;

// for문 사용
start = performance.now();
let visited1 = [];
for (let i = 0; i < n; i++) {
    visited1.push(0);
}
end = performance.now();
console.log(`for문 사용: ${end - start}ms`);

// fill 사용
start = performance.now();
let visited2 = Array(n).fill(0);
end = performance.now();
console.log(`fill 사용: ${end - start}ms`);